### PR TITLE
[DS-3667] Document fundamental persistence support classes.

### PR DESCRIPTION
**Pull request by mwoodiupui**
https://jira.duraspace.org/browse/DS-3667

### CODE REVIEW

**Comment by mwoodiupui** [comment]
Please tell me how to improve this.  I needed documentation _because I don't understand these classes_, so I may have missed or mistaken quite a bit.

**Comment by tdonohue** [comment]
This looks great to me so far, @mwoodiupui.  I'd rather see this get merged quickly, just cause some documentation/comments are better than nothing. If we find parts are misleading later, we can always clean them up.

**Review by tdonohue — APPROVED** [review]

File: dspace-api/src/main/java/org/dspace/core/DBConnection.java [hunk 1]
```diff
@@ -13,48 +13,121 @@
...
+     * @return the reloaded entity.
+     * @throws java.sql.SQLException passed through.
      */
     public <E extends ReloadableEntity> E reloadEntity(E entity) throws SQLException;
 
     /**
-     * Remove a DSpace object from the cache when batch processing a large number of objects.
-     * @param entity The DSpace object to reload
+     * Remove a DSpace object from the session cache when batch processing a
+     * large number of objects.
```
Comment (tdonohue) [inline dspace-api/src/main/java/org/dspace/core/DBConnection.java]: Minor note. Assuming `uncacheEntity` stays as-is, simply a tool to remove objects from session cache, we may want to add a warning here to note that "Objects removed from cache are not saved in any way. Therefore, if you have modified an object, you should be sure to commit() changes before calling this method"  (or something similar)

File: dspace-api/src/main/java/org/dspace/core/DBConnection.java [hunk 2]
```diff
@@ -13,48 +13,121 @@
...
+ *
+ * <p> This class should only be accessed by an enclosing {@link Context} object.
  *
  * @author kevinvandevelde at atmire.com
  * @param <T> class type
  */
 public interface DBConnection<T> {
 
+    /**
+     * Access to the underlying session object.
```
Comment (tdonohue) [inline dspace-api/src/main/java/org/dspace/core/DBConnection.java]: While I know you already clarified the meaning of "session" at the top of this class, I'm not sure if we should (re)clarify in this method that "session" is a persistence provider session (and has no relation to HTTPSession). 

If you think this is obvious (even to newbies), feel free to ignore this note.

**Review by terrywbrady — APPROVED** [review]

**Pull request merged**